### PR TITLE
49 init config

### PR DESCRIPTION
**Pull request by rpigu-i**
Created two new commands:

1. init - creates a capsule.json file with the project parameters 

2. deploy - deploys the capsule.json file

### CODE REVIEW

File: bin/capsule.js [hunk 1]
```diff
@@ -10,41 +10,160 @@ const chalk = require('chalk');
...
+const error_states = [
+  'CREATE_FAILED',
+  'DELETE_FAILED',
+  'UPDATE_FAILED',
+  'ROLLBACK_FAILED',
+  'UPDATE_ROLLBACK_FAILED'
+];
+
+const paths = {
+  base: `${__dirname}/../`,
```
Comment (grgur) [inline bin/capsule.js]: I'd go with
```js
base: path.resolve(__dirname, '../')
```

File: bin/capsule.js [hunk 2]
```diff
@@ -10,41 +10,160 @@ const chalk = require('chalk');
...
+ * @method writeConfigFile
+ *
+ * @param {Object} answers
+ *
+ * @return {void}
+ */
+const writeConfigFile = (answers) => {
+  fs.writeFile(paths.output_config, JSON.stringify(answers), function(err) {
+    if(err) {
+      return console.log(err);
```
Comment (grgur) [inline bin/capsule.js]: or `console.error(err)`

File: bin/capsule.js [hunk 3]
```diff
@@ -10,41 +10,160 @@ const chalk = require('chalk');
...
 
-
 //TODO - move code for assigning vars into a separate function
 //to allow code re-use.
 commander
   .command('init')
+  .description('Define the project parameters')
+  .action(async function (options) {
+    commander.type = 'init'
+    let generic_questions = await parseJsonConfig(`${paths.base}/${paths.base_config}`)
```
[the remark this answers is not in the document]
Comment (rpigu-i) [reply bin/capsule.js]: As per your previous comment, I've updated `const paths` to use:

```
base: path.resolve(__dirname, '../'),
```
So any reference to `path.base` now uses the `path.resolve` method. Is this what you had in mind?

File: bin/capsule.js (hunk 1 above)
Comment (rpigu-i) [reply bin/capsule.js]: Done!

File: bin/capsule.js (hunk 2 above)
Comment (rpigu-i) [reply bin/capsule.js]: Good spot. Have updated this to use the logIfVerbose function as well.

**Comment by rpigu-i** [comment]
@theneverstill I've updated some of the variables to be `const` rather than `let` as per your comment.

File: bin/capsule.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (rpigu-i) [reply bin/capsule.js]: Ahh OK gotcha, sorry misread the initial comment!

**Comment by rpigu-i** [comment]
@theneverstill  and @grgur  Have addressed the comments from above.

Please review again. Thanks!

**Comment by grgur** [comment]
i was faster

**Comment by rpigu-i** [comment]
@grgur haha OK, I had to push a second commit up - can you confirm again?

**Pull request merged**